### PR TITLE
Initial grunt work to update biome api

### PR DESCRIPTION
**Pull request by i509VCB**
**A list of the changes:**

All API biome methods require use of `RegistryKey<Biome>`. This is in line with vanilla now

`FabricBiomes#addSpawnBiome` is no longer needed since this is now a field on `Biome` itself. So it has been removed.

Replace the one instance of reflection in biome api with an accessor mixin.

Update (most?) of the mixins that aren't nuked completely by the update

Obvously this does not work yet and some internal logic needs to be changed a bit or looked at still

### CODE REVIEW

File: fabric-biomes-v1/src/main/java/net/fabricmc/fabric/impl/biome/InternalBiomeData.java [hunk 1]
```diff
@@ -109,97 +116,79 @@ public static void setOverworldRiverBiome(Biome primary, Biome river) {
...
+				picker.addBiome(BuiltinRegistries.BIOME.get(Biomes.PLAINS), 1);
+				picker.addBiome(BuiltinRegistries.BIOME.get(Biomes.FOREST), 1);
 			} else if (base == Biomes.DEEP_FROZEN_OCEAN) {
 				// Note: Vanilla Deep Frozen Oceans only have a 1/3 chance of having default hills.
 				// This is a clever trick that ensures that when a mod adds hills with a weight of 1, the 1/3 chance is fulfilled.
 				// 0.5 + 1.0 = 1.5, and 0.5 / 1.5 = 1/3.
 
-				picker.addBiome(Biomes.PLAINS, 0.25);
-				picker.addBiome(Biomes.FOREST, 0.25);
+				picker.addBiome(BuiltinRegistries.BIOME.get(Biomes.PLAINS), 0.25);
```
Comment (liach) [inline fabric-biomes-v1/src/main/java/net/fabricmc/fabric/impl/biome/InternalBiomeData.java:180]: concern this is not how you use builtin registries

**Review by liach — CHANGES_REQUESTED** [review]
No, we shouldn't have all these wrong ways of getting biome and bring in troubles in the future. Maybe you can add a static method in impl for getting a biome from a registry key for now, but getting from built in registry is definitely **not** the way to go.

File: fabric-biomes-v1/src/main/java/net/fabricmc/fabric/impl/biome/InternalBiomeData.java (hunk 1 above)
Comment (i509VCB) [reply fabric-biomes-v1/src/main/java/net/fabricmc/fabric/impl/biome/InternalBiomeData.java:180]: How then lol, don't complain and provide no solution.

**Comment by liach** [comment]
@i509VCB You wanted a solution. So there is `class_5504` per FabricMC/yarn#1660, and I wonder if that class allows you to get datapack biomes.

**Comment by modmuss50** [comment]
Ive had a real good go at trying to update this, making it compile isnt too bad, commenting out 2 places, but getting the mixins to be valid has truned out to be something of a major task. I think its almost worth dropping this API and starting again. I dont know a lot about how the biome gen works so that doesnt help either.

I do want to get https://github.com/FabricMC/fabric/pull/982 in somewhere as its very useful to a lot of mods.

**Comment by CheaterCodes** [comment]
I think this could be closed in favor of #1036 ?
At least we're definitely not gonna be able to salvage any of the old api.

**Comment by liach** [comment]
I don't think #1036 is in any means better than this.

**Comment by CheaterCodes** [comment]
actually this does much more than I thought it did. I need to look at this closer.

**Comment by i509VCB** [comment]
Closing this since #1036 has handled more of this

**Pull request closed without merging**